Lib/zipapp.py

```diff
@@ -134,8 +135,9 @@ def create_archive(source, target=None, interpreter=None, main=None):
         _write_file_prefix(fd, interpreter)
         with zipfile.ZipFile(fd, 'w') as z:
             for child in source.rglob('*'):
-                arcname = child.relative_to(source).as_posix()
-                z.write(child, arcname)
+                if include_file is None or include_file(pathlib.Path(child)):
```

Comment:
The ```include_file``` check should take ```arcfile``` as argument - specifically it should be passed a path that is relative to the source directory.